### PR TITLE
Develop/test docs

### PR DESCRIPTION
**Pull request by mpdesantis**
Just a bit of cleanup and brief READMEs in each directory. 
Sample program output added as a system test expectation.
Some minor printf modification just to unify the format between each subsystem's console output.
Did not format the printfs with newlines, since multithreaded console prints are hard to predict, did not know how to logically break apart. Maybe a TODO for later.

Thanks!

### CODE REVIEW

**Review by orselane — COMMENTED** [review]
Src changes: Print clarity, great

Test READMEs: should we separate it from the main readme? We do have a test section in there, but we should update it since we've changed the test folder quite a bit. If we keep it seperate, maybe just a single readme in the root test folder would be better to consolidate them? It just seems like a lot of documents to manage long term otherwise.

Expected output: I think we should keep a snippet, but describe what we're actually looking for. The output is not perfectly reproducible, but there is certain behavior we should be able to observe.

File: test/unit/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# README
+* Last Edited: 2024/02/03
+
+## Description
+Unit tests for relevant classes. 
+
+## Testing Notes
+* Only classes that expose public methods are unit tested.
```
Comment (orselane) [inline test/unit/README.md:8]: This note is really important, I would maybe go ahead and add it as an extra assumption in main readme's scope section.

Comment (mpdesantis) [reply test/unit/README.md:8]: will do now good buddy

Comment (mpdesantis) [reply test/unit/README.md:8]: i've added this to main readme on separate branch and will push after this PR is closed, put it in the test section tho and listed that we're using JUnit 5.8.1 as test framework (that's the version we use, and the version we're supposed to use from Assignment00 doc).

**Review by orselane — APPROVED** [review]

**Pull request merged**